### DISCUSSION TITLE
Generate thumbnails for all images and configure thumbnail size by theme

### DISCUSSION THREAD

**Discussion by NateWr** [pkp/pkp-lib#7969] [Proposals]
I'd like to see the thumbnail image size defined by the theme, rather than as an option in the settings. It's not really possible to design a theme that works well with any thumbnail size.

Default values could be set in `ThemePlugin.inc.php` with each theme overwriting it as necessary.

```
class ThemePlugin {
  private $thumbnailWidth = 125;
  private $thumbnailHeight = 250;
}

class MyThemePlugin extends ThemePlugin {
  private $thumbnailWidth = 150;
  private $thumbnailHeight = 300;
}
```

**Comment by NateWr** [comment]
@asmecher I might punt this to OMP 3.0's milestone since we're still a ways off from having a multi-theme environment where this would be really useful. What do you think?

**Comment by asmecher** [comment]
Yes, that's OK with me. Moving over to OJS 3.0.

**Comment by bozana** [comment]
Is this OJS 3.0 milestone?  @NateWr wrote OMP 3.0 not OJS 3.0 above :-)

**Comment by asmecher** [comment]
Hmm, perhaps not. I'm just not used to hearing anyone say "OMP 3.0".

**Comment by NateWr** [comment]
lol, this will probably get punted, but let's see how we feel after the meeting tonight.

**Comment by NateWr** [comment]
Punting.

**Comment by NateWr** [comment]
We don't even use thumbnails in OJS 3. Moved the milestone over to OMP, but it doesn't seem like a big priority there since there's not much theme work in place.

**Comment by jonasraoni** [comment]
Just copying my message from a discussion that we had recently.

- Maybe it’s worth adding an image optimization to serve better sizes (responsive images) and formats such as avif (https://aomediacodec.github.io/av1-avif/), webp, etc. on the fly... E.g. `/image.jpg?mobile=1&width=400` (could redirect the user to an static/cached resource), pain points would be invalidation (removing orphan/old images) + anti-abuse (maybe by defining allowed options/sizes). A garbage collector would be needed (https://github.com/pkp/pkp-lib/discussions/10274).
- WordPress has a theme define the sizes it uses. When an image is uploaded, it automatically generates those file sizes. When a theme is switched, the images are regenerated to match the new theme’s sizes. I’d like to do something like this for article/issue cover images, homepage image, and logo
- Lazy load content for free (https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading), would decrease the issues from the first item (a page which has lots of heavy images would cause some havoc on a first access)
